examples/using-gatsby-image/src/components/image-gallery.js

```diff
@@ -0,0 +1,103 @@
+import React from "react"
+import Img from "gatsby-image"
+import numeral from "numeral"
+
+import { options, scale } from "../utils/typography"
+import presets from "../utils/presets"
+
+const UnsplashMasonry = edges => (
+  <div
+    css={{
+      background: `white`,
+      margin: `0 -`,
+      [presets.Tablet]: {
+        margin: `0 0 0 calc(-${presets.offset}vw - ${presets.gutter.tablet}px)`,
+        padding: 100,
+        paddingRight: 0,
+      },
+      [presets.Desktop]: {
+        margin: `0 0 0 calc(-${presets.offset}vw - ${
+          presets.gutter.desktop
+        }px)`,
+      },
+    }}
+  >
+    <div
+      css={{
+        columnCount: 1,
+        columnGap: 0,
+        margin: `0 auto`,
+        [presets.Mobile]: {
+          columnCount: 2,
+        },
+        [presets.Tablet]: {
+          columnCount: 3,
+        },
+        "@media screen and (min-width: 1400px)": {
+          columnCount: 4,
+        },
+      }}
+    >
+      {edges.images.map((image, index) => (
+        <div
+          key={index}
+          css={{
+            border: `4px solid transparent`,
+            breakInside: `avoid`,
+            position: `relative`,
+            [presets.Mobile]: {
+              borderWidth: 8,
+            },
+            [presets.Desktop]: {
+              borderWidth: 12,
+            },
+            "& img": {
```

Comment:
Using the styled component approach will make this easier, too:

```js
const Component = styled('div')`
	background-color: red;

	img {
		border-radius: 2px;
	}
`;
```

Reply:
I wouldn't call that "easier", just different

Reply:
Sorry — "easier" is in reference to the syntax being less complex, but agreed that it's just different.

Reply:
When we decided on `styled`, did we also decide against [object styles](https://emotion.sh/docs/object-styles)?

Reply:
@fk yeah, we decided to prefer tagged template literals. Keep things as CSS-ey as we can to keep the APIs familiar to devs who maybe haven't done as much CSS-in-JS without giving up the benefits of CSS-in-JS.

Reply:
@fk First of all the page looks great :heart: 



I support @jlengstorf with using emotion full syntax with tagged template literals https://github.com/gatsbyjs/gatsby/pull/9377#discussion_r229269948
```
styled("div")`
...
`
```
There are a couple of ways to compose styles.
1. The `css` way

```
const baseStyle = css`
  flex-grow: 1;
  font-size: 0.85rem;
  margin: 0;
  color: ${colors.gray.lightCopy};
  padding: ${rhythm(0.5)} 0 0;
`
const Description = styled("p")`
  ${baseStyle};
  color: red;
```
2. Re-styling components way
```
const Description = styled("p")`
  flex-grow: 1;
  font-size: 0.85rem;
  margin: 0;
  color: ${colors.gray.lightCopy};
  padding: ${rhythm(0.5)} 0 0;
`

const DescriptionBig = styled(DescriptionBase)`
  font-size: 2rem;
`
```
3. merging css classes with `cx` method
https://emotion.sh/docs/cx

4. And of course using dynamic styles
https://emotion.sh/docs/composition#composing-dynamic-styles

I think that we should prefer the fourth and the second method. 

/cc @amberleyromo @jlengstorf @DSchau

Reply:
You can make styles as strings work but it feels like a big step back imo

Reply:
@KyleAMathews Sure, it could be that I do not spot some use cases when the objects work better.

Reply:
I think there are edge cases where styles work better, but overall moving to a JS abstraction over another language is a barrier for contributors who already know CSS.

My preference is to avoid logic in styles, and instead to have base styles that get extended where necessary:

```
const Button = styled('button')`
  background: red;
  border-radius: 0.5rem;
  color: white;
  font-size: 1.5rem;
`;

const BigButton = styled(Button)`
  font-size: 2.5rem;
`;
```

This makes it extremely clear that `BigButton` is an extension of `Button`, and doesn't require keeping how both JS and CSS work in your brain at the same time. 

Conversely, the JS version looks less like styles and (for my brain) is significantly harder to parse whether it's logic for the UI or logic for the styles:

```
const Button = styled('button')(props => ({
  background: 'red',
  borderRadius: '0.5rem',
  color: 'white',
  fontSize: props.isBig ? '2.5rem' : '1.5rem',
}));
```

It's also less clear _which_ button is being used in the second case, because the styles are now tied to the props and the result is hidden.

It's definitely going to come down to preference at the end of the day, but I feel pretty strongly that JS styles don't create enough benefit to justify the extra cognitive overhead.